lib/Dialect/ESI/ESIPasses.cpp

```diff
@@ -509,21 +516,25 @@ bool ESIPortsPass::updateFunc(HWModuleOp mod) {
     newOutputOperands.push_back(unwrap.getRawOutput());
     newResultTypes.push_back(chanTy.getInner()); // Raw data.
     newResultNames.push_back(oldResultName);
+    newResultLocs.push_back(oldResultLoc);
 
     newResultTypes.push_back(i1); // Valid.
     newOutputOperands.push_back(unwrap.getValid());
     newResultNames.push_back(appendToRtlName(oldResultName, "_valid"));
+    newResultLocs.push_back(oldResultLoc);
 
     newArgTypes.push_back(i1); // Ready func arg.
     newArgNames.push_back(appendToRtlName(oldResultName, "_ready"));
+    newArgLocs.push_back(oldResultLoc);
     updated = true;
   }
 
   // Append the ready list signals we remembered above.
   for (const auto &readySig : newReadySignals) {
     newResultTypes.push_back(i1);
-    newResultNames.push_back(readySig.second);
-    newOutputOperands.push_back(readySig.first);
+    newOutputOperands.push_back(std::get<0>(readySig));
```

Comment:
super nit: I generally prefer unboxing tuples in the loop. We upgraded the c++ version (IIRC since this code was written) to a version which supports unboxing.

Reply:
Its much better that way 👍